lib/semantics/check-do.cc

```diff
@@ -319,14 +319,42 @@ class DoConcurrentLabelEnforce {
   parser::CharBlock doConcurrentSourcePosition_{nullptr};
 };  // class DoConcurrentLabelEnforce
 
-using CS = std::vector<const Symbol *>;
+// Class for enforcing C1130
+class DoConcurrentVariableEnforce {
+public:
+  DoConcurrentVariableEnforce(
+      SemanticsContext &context, parser::CharBlock doConcurrentSourcePosition)
+    : context_{context}, doConcurrentSourcePosition_{
+                             doConcurrentSourcePosition} {}
 
-struct GatherSymbols {
-  CS symbols;
-  template<typename T> constexpr bool Pre(const T &) { return true; }
-  template<typename T> constexpr void Post(const T &) {}
-  void Post(const parser::Name &name) { symbols.push_back(name.symbol); }
-};
+  template<typename T> bool Pre(const T &) { return true; }
+  template<typename T> void Post(const T &) {}
+
+  // Check to see if the name is a variable from an enclosing scope
+  void Post(const parser::Name &name) {
+    if (const Symbol * symbol{name.symbol}) {
+      if (IsVariableName(*symbol)) {
+        const Scope &variableScope{symbol->owner()};
+        if (DoesScopeContain(&variableScope, GetBlockScope())) {
+          context_.Say(name.source,
+              "Variable '%s' from enclosing scope in a DEFAULT(NONE) DO "
+              "CONCURRENT, must appear in a locality-spec"_err_en_US,
+              name.source);
+        }
+      }
+    }
+  }
+
+private:
+  const Scope &GetBlockScope() {
+    return context_.FindScope(doConcurrentSourcePosition_);
+  }
+
+  SemanticsContext &context_;
+  parser::CharBlock doConcurrentSourcePosition_;
+};  // class DoConcurrentVariableEnforce
+
+using CS = std::vector<const Symbol *>;
```

Comment:
Why a `vector` instead of a `set`?

Reply:
I'm not sure.  That code was written before my time.  It doesn't look like the uses of ```CS``` require ordering.  I'll look into changing it.